system.c

```diff
@@ -176,6 +176,14 @@ uint8_t system_execute_line(char *line)
           else { report_ngc_parameters(); }
           break;
 
+        case 'P': // Set force sensitivity parameter. Avoids hastle of changing eeprom settings.
+          char_counter++;
+          if (!read_float(line, &char_counter, &parameter)){
```

Comment:
`read_float` converts the `char` version of the number that it reads from the serial port into a `float`. This function uses the `char` version of the decimal point as a delimiter between the whole number component and the fractional component and then stores it as a `float`. Thus if there is no decimal point, then the float is simply a whole number. `read_float` is located in `nuts_bolts.c` if you're interested.

`settings.force_sensor_level` , on the other hand, is a `uint8_t`, so there is an implicit cast on `parameter` from a `float` to `uint8_t`. It would probably be a good idea for me to make this an explicit cast for clarity.